cg/meta/upload/gisaid/gisaid.py

```diff
@@ -102,48 +136,59 @@ def build_gisaid_csv(self, gsaid_samples: List[GisaidSample], file_name: str) ->
-            gisaid_sample = GisaidSample(
-                family_id=family_id,
-                cg_lims_id=sample_id,
-                covv_subm_sample_id=sample.name,
-                submitter=self.gisaid_submitter,
-                fn=f"{family_id}.fasta",
-                covv_collection_date=self.lims_api.get_sample_attribute(
-                    lims_id=sample_id, key="collection_date"
-                ),
-                region=self.lims_api.get_sample_attribute(lims_id=sample_id, key="region"),
-                region_code=self.lims_api.get_sample_attribute(
-                    lims_id=sample_id, key="region_code"
-                ),
-                covv_orig_lab=self.lims_api.get_sample_attribute(
-                    lims_id=sample_id, key="original_lab"
-                ),
-                covv_orig_lab_addr=self.lims_api.get_sample_attribute(
-                    lims_id=sample_id, key="original_lab_address"
-                ),
-            )
-            gisaid_samples.append(gisaid_sample)
-        return gisaid_samples
+        log_file = Path(f"{self.gisaid_log_dir}/{family_id}.log")
+        if not log_file.parent.exists():
+            raise ValueError(f"Gisaid log dir: {self.gisaid_log_dir} doesnt exist")
+        return log_file
+
+    def append_log(self, temp_log: Path, gisaid_log: Path) -> None:
+        """appends temp log to gisaid log"""
+
+        with open(str(gisaid_log.absolute()), "r") as open_gisaid_log:
+            with open(str(temp_log.absolute()), "r") as open_temp_log:
+                gisaid_log_list: List = json.load(open_gisaid_log)
+                temp_log_list: List = json.load(open_temp_log)
+        gisaid_log_list.extend(temp_log_list)
+        with open(str(gisaid_log.absolute()), "w") as open_gisaid_log:
+            json.dump(gisaid_log_list, open_gisaid_log)
+        temp_log.unlink()
+
+    def file_to_hk(self, case_id: str, file: Path, tags: list):
+        version_obj: Version = self.housekeeper_api.last_version(case_id)
+        if not version_obj:
+            LOG.info("Family ID: %s not found in housekeeper", case_id)
+            raise HousekeeperVersionMissingError
+        self.housekeeper_api.add_file(version_obj=version_obj, tags=tags, path=str(file.absolute()))
+        self.housekeeper_api.commit()
+
+    def file_in_hk(self, case_id: str, tags: list) -> File:
+        version_obj: Version = self.housekeeper_api.last_version(case_id)
+        if not version_obj:
+            LOG.info("Family ID: %s not found in housekeeper", case_id)
+            raise HousekeeperVersionMissingError
+        file: File = self.housekeeper_api.files(version=version_obj.id, tags=tags).first()
+        return file
 
-    def upload(self, csv_file: Path, fasta_file: Path) -> None:
+    def upload_results_to_gisaid(self, files: UploadFiles) -> None:
         """Load batch data to GISAID using the gisiad cli."""
 
+        temp_log_file = "/tmp/gisaid_log"
```

Comment:
Should you not use a python generated temp file here?

Reply:
something like:
```
import tempfile
temp_log_file = tempfile.TemporaryFile(dir="/tmp/my_subdirectory", mode='"w+")
```

Reply:
👍